scriptworker/task.py

```diff
@@ -229,16 +230,15 @@ def guess_content_type(path):
     }
     args = [get_task_id(context.claim_task), get_run_id(context.claim_task),
             target_path, payload]
+    _compress_if_text_plain(path, payload['contentType'])
```

Comment:
On my machine, /etc/mime.types maps `asc` to `application/pgp-signature`. As a consequence, we currently don't gzip it. For example:
```sh
 curl -I https://public-artifacts.taskcluster.net/SwE-nbJ_RQalWmmEfMbCuA/0/public/chainOfTrust.json.asc
```

I agree this may not be the case on every machine. I see several solutions:

* Either we don't gzip `asc` file. In this case, we have to:
  * document `/etc/mime.types/` is extensively used, and different machines may provide different uploads
  * make the puppet module fill `/etc/mime.types` for us. For the record, our CentOS VMs don't have this file.
* Or we gzip files that are listed (hardcoded?) in scriptworker. At the moment, I'm thinking of files like: logs, json, asc, html/xml. If we go this way, here are the implications:
  * **Browsers** will be able to gunzip *some files*, which means:
    * If you want to use curl or wget, you have to pass extra args or pipe them (respectively `--compressed` and `wget ... | gunzip -c -`). Documenting that behavior would be nice.
    * If you want browser to understand every file, you have to force `Content-Type` to `text/plain`. This doesn't cause any issues once the file is downloaded. 

I checkout out some existing jobs listed on Treeherder. The majority of them just gzip live_backing.log. However, I saw some of them compressing html or json files.

I currently prefer the second solution, for 4 reasons:

1. We save space (and bandwidth) on S3, so the cost of hosting artifacts is lower.
2. Less unexpected behaviors to document. 
3. Every text file is handled the same.
4. Thanks to specifying `text/plain` for artifacts like `asc`, we make them readable from the browser (instead of downloadable, which happens on my computer). Thus, less steps to see the content of these files, faster dev/troubleshooting! 

What do you think @escapewindow